contributing/development/core_and_modules/internal_audio_architecture.rst

```diff
@@ -0,0 +1,236 @@
+.. _doc_internal_audio_architecture:
+
+Internal audio architecture
+===========================
+
+This page is a high-level overview of Godot 4's internal audio design.
+It does not apply to previous Godot versions.
+
+The goal of this page is to document design decisions taken to best suit
+:ref:`Godot's design philosophy <doc_best_practices_for_engine_contributors>`,
+while providing a starting point for new audio contributors.
+
+If you have questions about audio internals not answered here, feel free to
+ask in the ``#audio`` channel of the
+`Godot Contributors Chat <https://chat.godotengine.org/channel/audio>`__.
+
+
+.. note::
+
+    If you have difficulty understanding concepts on this page,
+    it is recommended to go through an audio blog such as
+    `Audio Sample Rate and Bit Depth <https://www.izotope.com/en/learn/digital-audio-basics-sample-rate-and-bit-depth.html>`__,
+    `How Sound Works <https://audiouniversityonline.com/audio-basics-how-sound-works/>`__, and
+    `An Introduction to Audio Electronics <https://www.allaboutcircuits.com/technical-articles/introduction-audio-electronics-sound-microphones-speakers-amplifiers>`__.
+
+
+.. _doc_internal_audio_architecture_drivers:
+
+Audio Drivers
+-------------
+
+Supported
+- Dummy
+- linuxBSD
+
+   - Alsa
+   - PulseAudio
+   - Alsa MIDI
+
+- Windows + UWP
+
+   - WASAPI
+   - XAudio2
```

Comment:
XAudio2 isn't compiled by default, and [it currently doesn't build](https://github.com/godotengine/godot/issues/75109#issuecomment-1724273758). It may be removed from the source tree at some point.